File: TypeScript/Codewars/src/resolutions/Balanced-Number-Special-Numbers-Series-1.ts
export function balancedNum(number: number): string {
	const arrayValues = String(number).split("").map(Number);
	const length = arrayValues.length;

	if (length <= 2) return "Balanced";

	const halfLength = Math.floor(length / 2);
	const isEven = length % 2 === 0;

	const sumLeft = arrayValues.slice(0, halfLength - (isEven ? 1 : 0)).reduce((a, b) => a + b, 0);
	const sumRight = arrayValues.slice(halfLength + 1).reduce((a, b) => a + b, 0);

	return sumLeft === sumRight ? "Balanced" : "Not Balanced";
}

File: Csharp/CodeWars/src/Completed/BalancedNumberSpecialNumbersSeries1.cs
namespace CodeWars.Completed;

public class BalancedNumberSpecialNumbersSeries1
{
    public static string BalancedNumber(int number)
    {
        int[] arrayValues = [.. number.ToString().Select(x => x - '0')];
        int length = arrayValues.Length;

        if (length <= 2) return "Balanced";

        int halfLength = length / 2;
        bool isEven = length % 2 == 0;

        int leftEnd = isEven ? halfLength - 1 : halfLength;
        int rightStart = isEven ? halfLength + 1 : halfLength + 1;

        int sumLeft = arrayValues[..leftEnd].Sum();
        int sumRight = arrayValues[rightStart..].Sum();

        return sumLeft == sumRight ? "Balanced" : "Not Balanced";
    }
}